overlord/snapstate/backend/setup.go

```diff
@@ -111,6 +114,23 @@ func (b Backend) RemoveSnapFiles(s snap.PlaceInfo, typ snap.Type, meter progress
 	return nil
 }
 
+func (b Backend) RemoveSnapDir(s snap.PlaceInfo, otherInstances bool) error {
+	mountDir := s.MountDir()
+
+	// try to remove the mount point base dir, failure is ok here
+	snapName, instanceKey := snap.SplitInstanceName(s.InstanceName())
+	if instanceKey != "" {
+		// always ok to remove instance specific one
+		os.Remove(filepath.Dir(mountDir))
```

Comment:
We ignore failures here a comment about the rational for this would be great (same for the os.Remove a few lines below).